tests/unittests/test_number_transform.py

```diff
@@ -1,25 +1,75 @@
+        value = 1234
+        transformed_data = transform_sheet_number_data("1.23E+03", value, sheet_title='test-sheet', col_name='test-column', col_letter='col', row_num=1, col_type='numberType')
+        self.assertIsInstance(transformed_data, int)
+        self.assertEqual(transformed_data, 1234)
+
+    def test_number_transform_int_US_format_value_as_int(self):
+        """Verify that US format int values falls back as type int"""
+        value = 1234
+        transformed_data = transform_sheet_number_data("1,234", value, sheet_title='test-sheet', col_name='test-column', col_letter='col', row_num=1, col_type='numberType')
+        self.assertIsInstance(transformed_data, int)
+        self.assertEqual(transformed_data, 1234)
+
     def test_number_transform_float_value_as_float(self):
         """Verify that float values falls back as type float"""
-        value = 1.0
-        transformed_data = transform_sheet_number_data(value, sheet_title='test-sheet', col_name='test-column', col_letter='col', row_num=1, col_type='numberType')
+        value = 1.1
+        transformed_data = transform_sheet_number_data("1.1", value, sheet_title='test-sheet', col_name='test-column', col_letter='col', row_num=1, col_type='numberType')
+        self.assertIsInstance(transformed_data, float)
+        self.assertEqual(transformed_data, 1.1)
+
+    def test_number_transform_float_exponential_value_as_float(self):
+        """Verify that exponential float values falls back as type float"""
+        value = 5e-16
+        transformed_data = transform_sheet_number_data("5.00E-16", value, sheet_title='test-sheet', col_name='test-column', col_letter='col', row_num=1, col_type='numberType')
         self.assertIsInstance(transformed_data, float)
-        self.assertEqual(transformed_data, 1.0)
\ No newline at end of file
+        self.assertEqual(transformed_data, 5e-16)
+
+    def test_number_transform_float_US_format_value_as_float(self):
+        """Verify that US format float values falls back as type float"""
+        value = 1234.1
+        transformed_data = transform_sheet_number_data("1,234.1", value, sheet_title='test-sheet', col_name='test-column', col_letter='col', row_num=1, col_type='numberType')
+        self.assertIsInstance(transformed_data, float)
+        self.assertEqual(transformed_data, 1234.1)
+
+    def test_number_transform_datetime_value_as_string(self):
+        """Verify that datetime values falls back as type string"""
+
+        datetime_type = transform_sheet_number_data("01/01/2022 0:00:00", 44562, "test_sheet", "Number Column", "A", 4, "numberType")
+
+        self.assertEqual(datetime_type, "01/01/2022 0:00:00")
+        self.assertIsInstance(datetime_type, str)
+
+    def test_number_transform_time_value_as_string(self):
+        """Verify that time values falls back as type string"""
+
+        time_type = transform_sheet_number_data("5:00 PM", 0.7083333333333334, "test_sheet", "Number Column", "A", 5, "numberType")
```

Comment:
@harshpatel4crest change the variable name to `time_expected_value`

Reply:
Updated the variable name.